src/pages/index.js

```diff
@@ -58,32 +119,47 @@ const defautlFormConfig = {
 
 const editFormValidator = new FormValidator(defautlFormConfig, profilePencilModal);
 const addCardFormValidator = new FormValidator(defautlFormConfig, addCardModal);
+const editAvatarFormValidator = new FormValidator(defautlFormConfig, profileAvatarModal);
 
 editFormValidator.enableValidation();
 addCardFormValidator.enableValidation();
+editAvatarFormValidator.enableValidation();
 
 // Render Initial Cards from Section class
-const cardSection = new Section({
-  items: initialCards,
-  renderer: (cardData) => {
-     const cardElement = createCard(cardData);
-    cardSection.addItem(cardElement);
-  }
-}, '.cards__content');
 
-cardSection.renderItems();
-
-// UserInfo 
+let cardSection
+
+Promise.all([api.getCurrentUser(), api.getInitialCards()])
+  .then(([data, InitialCards]) => {
+    userInfo.setUserInfo({ title: data.name, description: data.about});
+    userInfo.setAvatar({avatar: data.avatar});
+    cardSection = new Section({
+      items: InitialCards.reverse(),
+      renderer: (cardData) => {
+         const cardElement = createCard(cardData);
+        cardSection.addItem(cardElement);
+      }
+    }, '.cards__content');
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(`Error: ${err}`)
+  });
 
-const userInfo = new UserInfo({
-  title: '.profile__title',
-  description: '.profile__description'
-});
 
 // Edit Profile Form 
 
 const profileEditPopup = new PopupWithForm('#profile-pencil-modal', (formData) => {
-   userInfo.setUserInfo(formData);
+  profileEditPopup.submitButtonState(true)
+  api
+    .updateProfile(formData)
+    .then((data) => {
+      userInfo.setUserInfo({ title: data.name, description: data.about});
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`)
+    })
+    .finally(() => profileEditPopup.submitButtonState(false));
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```